src/hotspot/share/gc/z/zRelocate.cpp

```diff
@@ -74,6 +74,7 @@ static uintptr_t relocate_object_inner(ZForwarding* forwarding, uintptr_t from_a
   // Copy object
   ZUtils::object_copy_disjoint(from_addr, to_addr, size);
 
+  OrderAccess::release();
```

Comment:
Could you please replace the calls to `OrderAccess::release()` with a single call here instead:

```--- a/src/hotspot/share/gc/z/zForwarding.inline.hpp
+++ b/src/hotspot/share/gc/z/zForwarding.inline.hpp
@@ -136,6 +136,8 @@ inline uintptr_t ZForwarding::insert(uintptr_t from_index, uintptr_t to_offset,
   const ZForwardingEntry new_entry(from_index, to_offset);
   const ZForwardingEntry old_entry; // Empty
 
+  OrderAccess::release();
+
   for (;;) {
     const ZForwardingEntry prev_entry = Atomic::cmpxchg(entries() + *cursor, old_entry, new_entry, memory_order_release);
     if (!prev_entry.populated()) {
```

While keeping the calls close to the copy code has some value, I think I'd prefer to have it in a single place, and close to the corresponding acquire in `at()`.

Reply:
I updated my comment above, since it had a typo (used `memory_order_release` instead of `memory_order_relaxed`).

Reply:
OK, it's better to add the fence in one single place.

Reply:
> I updated my comment above, since it had a typo (used `memory_order_release` instead of `memory_order_relaxed`).

I have replaced it. Thanks for your advice.